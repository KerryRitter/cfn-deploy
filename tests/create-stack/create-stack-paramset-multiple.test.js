const mockCreateStack = require('../mocks/createStack');
const lib = require('../..');
const stringParams = require('../params/string-params.json');


describe('cfn-deploy', () => {
  beforeAll(() => mockCreateStack.beforeAll());

  it('should successfully create new stack', (done) => {
    let paramsArray = [
      './tests/params/array-params.json',
      './tests/params/object-cp-params.json',
      './tests/params/object-params.json',
      {
        S3BucketName: 'from-object-params',
        DummyParam: 'dummy-param',
        OverwritableParam: 'from-object-params',
        ParamSource: 'plain-object',
      },
    ];
    paramsArray = paramsArray.concat(stringParams);
    const events = lib({
      region: 'us-east-1',
      stackname: 'new-stack',
      template: './tests/templates/params-template.yaml',
      parameters: paramsArray,
    });
    events.on('ERROR', (err) => done(err));
    events.on('COMPLETE', (data) => {
      expect(data).toMatchSnapshot();
    });
    events.on('FINALLY', done);
  });

  afterAll(() => mockCreateStack.afterAll());
});
