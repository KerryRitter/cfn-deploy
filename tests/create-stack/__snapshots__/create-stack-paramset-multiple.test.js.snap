// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cfn-deploy should successfully create new stack 1`] = `
Object {
  "Capabilities": Array [],
  "ChangeSetName": "cfn-deploy",
  "ChangeSetType": "CREATE",
  "Description": "Created with cfn-deploy",
  "Parameters": Array [
    Object {
      "ParameterKey": "S3BucketName",
      "ParameterValue": "from-array-params-file",
    },
    Object {
      "ParameterKey": "DummyParam",
      "ParameterValue": "dummy-param",
    },
    Object {
      "ParameterKey": "OverwritableParam",
      "ParameterValue": "from-array-params-file",
    },
    Object {
      "ParameterKey": "ParamSource",
      "ParameterValue": "array-params.json",
    },
    Object {
      "ParameterKey": "S3BucketName",
      "ParameterValue": "from-object-cp-params-file",
    },
    Object {
      "ParameterKey": "DummyParam",
      "ParameterValue": "dummy-param",
    },
    Object {
      "ParameterKey": "OverwritableParam",
      "ParameterValue": "from-object-cp-params-file",
    },
    Object {
      "ParameterKey": "ParamSource",
      "ParameterValue": "object-cp-params.json",
    },
    Object {
      "ParameterKey": "S3BucketName",
      "ParameterValue": "from-object-params-file",
    },
    Object {
      "ParameterKey": "DummyParam",
      "ParameterValue": "dummy-param",
    },
    Object {
      "ParameterKey": "OverwritableParam",
      "ParameterValue": "from-object-params-file",
    },
    Object {
      "ParameterKey": "ParamSource",
      "ParameterValue": "object-params.json",
    },
    Object {
      "ParameterKey": "S3BucketName",
      "ParameterValue": "from-object-params",
    },
    Object {
      "ParameterKey": "DummyParam",
      "ParameterValue": "dummy-param",
    },
    Object {
      "ParameterKey": "OverwritableParam",
      "ParameterValue": "from-object-params",
    },
    Object {
      "ParameterKey": "ParamSource",
      "ParameterValue": "plain-object",
    },
    Object {
      "ParameterKey": "S3BucketName",
      "ParameterValue": "from-string-params-file",
    },
    Object {
      "ParameterKey": "DummyParam",
      "ParameterValue": "dummy-param",
    },
    Object {
      "ParameterKey": "OverwritableParam",
      "ParameterValue": "from-string-params-file",
    },
    Object {
      "ParameterKey": "ParamSource",
      "ParameterValue": "string-params.json",
    },
  ],
  "StackName": "new-stack",
  "Tags": Array [],
  "TemplateBody": "Parameters:
  S3BucketName:
    Description: \\"S3 Bucket name\\"
    Type: String
  DummyParam:
    Description: \\"Dummy parameter\\"
    Type: String
  OverwritableParam:
    Description: \\"Parameter for testing overrides\\"
    Type: String
  ParamSource:
    Description: \\"Source where the parameters came from\\"
    Type: String

Resources:
  S3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: !Sub \\"\${AWS::StackName}-\${S3BucketName}\\"
",
}
`;

exports[`cfn-deploy should successfully create new stack 2`] = `
Object {
  "StackStatus": "REVIEW_IN_PROGRESS",
}
`;
